cmd/tools/mise.go

```diff
@@ -0,0 +1,385 @@
+	)
+
+	miseCmd.Flags().Bool(
+		"skip-deps-download",
+		false,
+		"Skip downloading the binaries",
+	)
+
+	miseCmd.Flags().Bool(
+		"revert",
+		false,
+		"Remove furyctl-managed tools from mise.toml instead of adding them",
+	)
+
+	miseCmd.Flags().Bool(
+		"force",
+		false,
+		"Skip confirmation prompt when reverting tools",
+	)
+
+	return miseCmd
+}
+
+// updateMiseConfig creates or updates the mise.toml file with furyctl tools.
+func updateMiseConfig(tools []ToolInfo) error {
+	const miseFile = "mise.toml"
+
+	// Load existing configuration or create new one.
+	config, err := loadMiseConfig(miseFile)
+	if err != nil {
+		return fmt.Errorf("failed to load existing mise config: %w", err)
+	}
+
+	// Update tools section with absolute paths.
+	for _, tool := range tools {
+		// Get absolute path to the tool's directory (not the binary itself).
+		// Mise expects the directory to contain a binary with the same name as the tool.
+		toolDir := filepath.Dir(tool.BinaryPath)
+		absPath, err := filepath.Abs(toolDir)
+		if err != nil { //nolint:wsl // gofumpt and wsl disagree on formatting
+			return fmt.Errorf("failed to get absolute path for %s: %w", tool.Name, err)
+		}
+
+		// Use mise's path: syntax to specify exact tool location.
+		config.Tools[tool.Name] = "path:" + absPath
+	}
+
+	// Write updated configuration back to file.
+	if err := saveMiseConfig(miseFile, config); err != nil {
+		return fmt.Errorf("failed to save mise config: %w", err)
+	}
+
+	return nil
+}
+
+// loadMiseConfig loads existing mise.toml file or returns empty config.
+func loadMiseConfig(filename string) (*MiseConfig, error) {
+	config := &MiseConfig{
+		Tools: make(map[string]string),
+		Other: make(map[string]any),
+	}
+
+	// Check if file exists.
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		// File doesn't exist, return empty config.
+		return config, nil
+	}
+
+	// Read existing file.
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s: %w", filename, err)
+	}
+
+	// Parse existing TOML, preserving all sections.
+	var rawConfig map[string]any
+	if err := toml.Unmarshal(data, &rawConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %w", filename, err)
+	}
+
+	// Extract tools section if it exists.
+	if toolsSection, exists := rawConfig["tools"]; exists {
+		if toolsMap, ok := toolsSection.(map[string]any); ok {
+			for k, v := range toolsMap {
+				if strVal, ok := v.(string); ok {
+					config.Tools[k] = strVal
+				}
+			}
+		}
+
+		delete(rawConfig, "tools") // Remove from other sections.
+	}
+
+	// Store all other sections to preserve them.
+	config.Other = rawConfig
+
+	return config, nil
+}
+
+// saveMiseConfig saves the mise configuration to file.
+func saveMiseConfig(filename string, config *MiseConfig) error {
+	// Combine tools section with other sections.
+	// This intermediate map is used to preserve the order of the sections in the final TOML file,
```

Comment:
I don't think order is being preserved, my mise.toml gets sorted completed different after using the `tools mise` command, it even breaks some vars that depend on the previous one.

Reply:
Also, it may be a good idea to run `mise fmt` after saving the file :)